### PR TITLE
Telco 219 integrate orc8r with prometheus and prometheus configurer

### PR DESCRIPTION
**Pull request by Gmerold**
# Description

This PR implements integration of Magma Orc8r with Prometheus and Prometheus Configurer.

# Checklist:

- [X] My code follows the [style guidelines](/CONTRIBUTING.md) of this project
- [X] I have performed a self-review of my own code
- [X] I have made corresponding changes to the documentation
- [X] I have added tests that validate the behaviour of the software
- [X] I validated that new and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: orchestrator-bundle/orc8r-metricsd-operator/src/charm.py [hunk 1]
```diff
@@ -66,25 +65,58 @@ def __init__(self, *args):
...
+            f'prometheusConfigServiceURL: "{self._prometheus_configurer_url}/v1"\n'
             f'alertmanagerConfigServiceURL: "{self.ALERTMANAGER_CONFIGURER_URL}/v1"\n'
             '"profile": "prometheus"\n'
         )
         self._orc8r_base.container.push(f"{self.BASE_CONFIG_PATH}/metricsd.yml", metricsd_config)
 
+    @property
+    def _prometheus_url(self) -> str:
+        prometheus_service_name = self.model.get_relation("prometheus-k8s").app.name  # type: ignore[union-attr]  # noqa: E501
+        return f"http://{prometheus_service_name}:9090"
```
Comment (ghislainbourgeois) [inline orchestrator-bundle/orc8r-metricsd-operator/src/charm.py:212]: Shouldn't the port be provided by the relation also?

File: orchestrator-bundle/orc8r-metricsd-operator/src/charm.py [hunk 2]
```diff
@@ -66,25 +65,58 @@ def __init__(self, *args):
...
+
+    @property
+    def _prometheus_configurer_url(self) -> str:
+        prometheus_configurer_service_name = self.model.get_relation(
+            "prometheus-configurer-k8s"
+        ).app.name  # type: ignore[union-attr]
+        return f"http://{prometheus_configurer_service_name}:9100"
```
Comment (ghislainbourgeois) [inline orchestrator-bundle/orc8r-metricsd-operator/src/charm.py:225]: Shouldn't the port be provided by the relation also?

File: orchestrator-bundle/orc8r-metricsd-operator/src/charm.py [hunk 3]
```diff
@@ -66,25 +65,58 @@ def __init__(self, *args):
...
+
+    @property
+    def _alertmanager_url(self) -> str:
+        alertmanager_service_name = self.model.get_relation("alertmanager-k8s").app.name  # type: ignore[union-attr]  # noqa: E501
+        return f"http://{alertmanager_service_name}:9093"
```
Comment (ghislainbourgeois) [inline orchestrator-bundle/orc8r-metricsd-operator/src/charm.py:236]: Shouldn't the port be provided by the relation also?

File: orchestrator-bundle/orc8r-metricsd-operator/src/charm.py [hunk 4]
```diff
@@ -66,25 +65,58 @@ def __init__(self, *args):
...
+    @property
+    def _relations_created(self) -> bool:
+        """Checks whether required relations are created."""
+        if missing_relations := [
+            relation
+            for relation in self.EXTERNAL_RELATIONS
+            if not self.model.get_relation(relation)
+        ]:
+            msg = f"Waiting for relation(s) to be created: {', '.join(missing_relations)}"
+            self.unit.status = BlockedStatus(msg)
```
Comment (ghislainbourgeois) [inline orchestrator-bundle/orc8r-metricsd-operator/src/charm.py:251]: We set `BlockedStatus` here but we never set `ActiveStatus`?

**Review by ghislainbourgeois — CHANGES_REQUESTED** [review]

File: orchestrator-bundle/bundle-local-federated.yaml [hunk 5]
```diff
@@ -239,6 +244,8 @@ relations:
   - postgresql-k8s:db
 - - orc8r-bootstrapper
   - orc8r-certifier
+- - orc8r-certifier
+  - tls-certificates-operator
```
Comment (gruyaume) [inline orchestrator-bundle/bundle-local-federated.yaml:248]: Why is this added in this PR?

Comment (Gmerold) [reply orchestrator-bundle/bundle-local-federated.yaml:248]: I've seen `tls-certificates-operator` charm added to this bundle, but the relation was missing. Was that intentional?

File: orchestrator-bundle/orc8r-metricsd-operator/src/charm.py (hunk 1 above)
Comment (Gmerold) [reply orchestrator-bundle/orc8r-metricsd-operator/src/charm.py:212]: I've added TODOs to get the port number from the relation data once it's available there.

File: orchestrator-bundle/orc8r-metricsd-operator/src/charm.py (hunk 3 above)
Comment (Gmerold) [reply orchestrator-bundle/orc8r-metricsd-operator/src/charm.py:236]: I've added TODOs to get the port number from the relation data once it's available there.

File: orchestrator-bundle/orc8r-metricsd-operator/src/charm.py (hunk 2 above)
Comment (Gmerold) [reply orchestrator-bundle/orc8r-metricsd-operator/src/charm.py:225]: I've added TODOs to get the port number from the relation data once it's available there.

File: orchestrator-bundle/orc8r-metricsd-operator/src/charm.py (hunk 4 above)
Comment (Gmerold) [reply orchestrator-bundle/orc8r-metricsd-operator/src/charm.py:251]: New implementation doesn't use the `orc8r_base` lib anymore and `ActiveStatus` is now set by the charm after configuring pebble layer.

**Review by ghislainbourgeois — APPROVED** [review]

**Review by gruyaume — APPROVED** [review]

**Pull request merged**